### ISSUE TITLE
local_assembly ??

### ISSUE THREAD

**Issue by pippo1990** [Andre-lab/evodock#4]
I am trying to figure out how evodock works.

Could be that I am mistaken and didnt get the real pourpose of the tool. I am trying to use:

> Local assembly - Predicting assembly structure from multiple backbones and starting positions:

   python ./evodock.py configs/symmetric/local_assembly.ini

**Comment by pippo1990** [comment]
mmmh think I got it wrong .. from  Accurate prediction of protein assembly structure by combining AlphaFold and symmetrical docking

> Selection of benchmark structures

The overall selection process for the cubic structure benchmark is described in Fig. [6](https://www.nature.com/articles/s41467-023-43681-6#Fig6). First a list of homomeric tetrahedral, octahedral, and icosahedral assemblies with 12, 24, and 60 chains, respectively, and with a resolution better than 4 Å was compiled from the Protein Data Bank[34](https://www.nature.com/articles/s41467-023-43681-6#ref-CR34)

**Comment by pippo1990** [comment]
A last question:

in 1STM.symm I can read :
.....
symmetry_name /home/shared/databases/SYMMETRICAL/I/unrelaxed/native/../../idealized/symdef/native/1STM.symm
....


Any chance I could build this file myself starting from 1STM.cif . ? (i.e command line / or script)

I installed your packege so I think I 've have PyRosetta 4 installed too

**Comment by MadsJeppesen** [comment]
Hi pippo1990,

subunits in the [Inputs] flag refers to a directory containing all pdb files (with different backbones) you wish to use in EvoDOCK. T>1 and heteromeric structures are untested but should be doable by combining the individual chains in a subunit file into a single chain. This trickers EvoDOCK to think it is a continuous chain and is T=1 and it will not move the constituent chains individually. 

It seems like you found the answer on how to generate a symmetry file (*.symm) as 1STM.symm is. As you mentioned you have to use an external library (cubicsym) to generate the symmetry file.

**Comment by pippo1990** [comment]
> Hi pippo1990,
> 
> subunits in the [Inputs] flag refers to a directory containing all pdb files (with different backbones) you wish to use in EvoDOCK. T>1 and heteromeric structures are untested but should be doable by combining the individual chains in a subunit file into a single chain. This trickers EvoDOCK to think it is a continuous chain and is T=1 and it will not move the constituent chains individually.
> 
> It seems like you found the answer on how to generate a symmetry file (*.symm) as 1STM.symm is. As you mentioned you have to use an external library (cubicsym) to generate the symmetry file.

yep went that way but I can create of course a monomer faking the AU of 2ms2 phage  (all 3 chains as A) but now I am at loss trying to get a symm file of it

**Comment by pippo1990** [comment]
> > Hi pippo1990,
> > subunits in the [Inputs] flag refers to a directory containing all pdb files (with different backbones) you wish to use in EvoDOCK. T>1 and heteromeric structures are untested but should be doable by combining the individual chains in a subunit file into a single chain. This trickers EvoDOCK to think it is a continuous chain and is T=1 and it will not move the constituent chains individually.
> > It seems like you found the answer on how to generate a symmetry file (*.symm) as 1STM.symm is. As you mentioned you have to use an external library (cubicsym) to generate the symmetry file.
> 
> yep went that way but I can create of course a monomer faking the AU of 2ms2 phage (all 3 chains as A) but now I am at loss trying to get a symm file of it
> 
> I mean its not hard to change the chains name all to As, but how to generate a meaningfull _pdbx_struct_oper_list (https://mmcif.wwpdb.org/dictionaries/mmcif_pdbx_v50.dic/Categories/pdbx_struct_oper_list.html) loop_ for the .cif file to be feed to cubicsymm ?

Just copied and pasted original one the script (cubicsymm) completed with no error  and got symm file , I'll test the generate full assembly too to see if I get the entire capsid correct then I guess I'll try evodeock with that .symm file

**Comment by MadsJeppesen** [comment]
It should be possible to score with a radius term inside EvoDOCK and it makes sense if you have confidence on the approximate radius. However, I have seen scattered subunits as well in my own simulations and they are a results of EvoDOCK not being able to find low energy interfaces in certain trajectories. Im confident if you run more simulations you would see better packed subunits were the radius also make sense. How many simulations have you run? You should likely run 10-50 runs to get reasonable results depending on your system. Let me know how it goes with the created symmetry file and I can assist you further if you run into any issues.

**Comment by pippo1990** [comment]
> It should be possible to score with a radius term inside EvoDOCK and it makes sense if you have confidence on the approximate radius. However, I have seen scattered subunits as well in my own simulations and they are a results of EvoDOCK not being able to find low energy interfaces in certain trajectories. Im confident if you run more simulations you would see better packed subunits were the radius also make sense. How many simulations have you run? You should likely run 10-50 runs to get reasonable results depending on your system. Let me know how it goes with the created symmetry file and I can assist you further if you run into any issues.

I just load the results and use Biopython to calculate the radius starting from Entity.center_of_mass(geometric=True) [https://biopython.org/docs/dev/api/Bio.PDB.Entity.html#Bio.PDB.Entity.Entity.center_of_mass] and getting the most distant residue/atom with Euclidean distance kind of very long and slow calculation, but works.

I have very limited computational resources so it take ages to get just 5  .cif results I am doing it as a hobby. Kind of got nice results for 1STM (none of which matches the original structure) but I am failing to get nice result using 2ms2 as template:

Just changing all 3 AU subunits A,B,C to chain A of a modified (some of the loops where deleted) 2ms2.cif and calculate .symm adding the loop_ pdbx_struct_oper_list to my pymol all A 2ms2 I am failing to get plausible results, Dont understand a lot about symmetry and so on but its like the results obtained try to get an icosahedral like particle out of the AU without dimerizing the subunits. In 2ms2 you get A,B,C monomers in AU that then create an Icosahedron of AB , CC dimers (60 A,B,C  for 180 total monomers) this should be in the symmetri matrix I collate to the modded all A cif , I am not sure of my .ini file , maybe I just miss some of the needed paramteters fur such kind of capsids [sorry to take your time , hope it is clear what I am trying to accomplish: get 2ms2.cif --> delete some bits of each monomer (all the 3 monomer have same deletion) --> trying  to recreate the final capsid ; basic idea is to trying to prove you can modify each one of the 3 monomers in some way to incorporate 3 different tags , next move will be to modify each of the monomer in a different way, I know is far fetched because of the way such capsids assemble (first they dimerize creating nucleation points and so on ) but I was fantasizing about mixing the 3 kinds of monomer toghether and some how rebuild the 3 tags decorated entire capsid]

**Comment by pippo1990** [comment]
more on this I wanted to test the : 
```
[Docking]
type=GlobalFromMultimer
```
in evodock .init feeding the software with the 3 separate monomers of 2ms2.cif as subunits 
```
[Inputs]
subunits=./2ms2_A_B_C_monomers/
```
and using the .symm file obtained as described above , same results : 

![Captureexp2](https://github.com/user-attachments/assets/92d818b3-8118-409b-a2ad-fac2dad3910a)

then I realized perhaps I need a different .symm file so I turned to cubicsym (https://github.com/Andre-lab/cubicsym)

to generate it (as done for the above) with:

`python ./cubicsym/scripts/cubic_to_rosetta.py --structures 22768/2ms2.cif --symmetry I --output_generated_structure --output_generated_structure_outpath 2ms2_cubicsymm_test/outputs --overwrite `

using the original 2ms2.cif as input but getting : 

a lot of `No symmetry dectected`

and an error that stop the script : 

```
raise NotImplemented(f"Code does not yet support ')(' combinations of operations such as {symmetry_operations}")
TypeError: 'NotImplementedType' object is not callable
```

I'll try to add --hf1 --hf2 .....  to see if I can progeress more

**Comment by MadsJeppesen** [comment]
Thank you for describing your approach. It makes sense to recereate the extract symmetry from 2MS2 with the symmetry script in cubicsym with the the goal you are trying to accomplish. Also to use a higher T number than 1, you would have to combine the chains as you have done. For local refinement I would use:

```
[Docking]
type=Local
```

Instead. 

You can check if it makes sense fast with the following options:

```
[DE]
popsize=4
maxiter=1
```

It should not move a lot in that case. Let me know if you have success. If the issue of having subunits out in space persist can you upload all your input files? In that case I can check what is going on in detail.

**Comment by pippo1990** [comment]
as per in my previous post (https://github.com/Andre-lab/evodock/issues/4#issuecomment-2325024980)§  my best results is :

![Captureexp3](https://github.com/user-attachments/assets/eefc06ee-e452-4d79-8c56-5b801be673e0)


got it with (
[local_recapitulation_mine_2ms2.ini.txt](https://github.com/user-attachments/files/16976289/local_recapitulation_mine_2ms2.ini.txt)
   ) : 
```

[Docking]
type=Local

[Inputs]
single=./2ms2_modded_chain_ABC_as_A_segi_too_SUPERPOSED_prepack.pdb
symdef_file=./2ms2_all_A_modded_sym_added.symm


[Outputs]
output_path=tests/outputs/local_recapitulation/
output_pdb=True


[Bounds]
bounds=5,18,5,40,40,40
allow_flip=True

[DE]
scheme=RANDOM
popsize=4
mutate=0.1
recombination=0.7
maxiter=3
local_search=symshapedock
slide=true
selection=interface

[RosettaOptions]
initialize_rigid_body_dofs=true

```

**Comment by pikkazz** [comment]
Hello,   @pippo1990 pippo. In the new version, evodock doesn't work and gets stuck at symmetry_type. This happens whether it's the provided examples or the ones you gave. However, it runs normally as long as symmetry_type is not used. So, I was wondering if you could send me a copy of your evodock file? Thank you very much.
